crates/oas3/src/error.rs

```diff
@@ -15,6 +15,7 @@ pub enum Error {
 
     /// YAML error.
     #[display("YAML error")]
+    #[cfg(feature = "yaml_spec")]
     Yaml(serde_yaml::Error),
```

Comment:
feature guarding enum variants is a no-go for me, i'll get this solved another way

Reply:
Still open to feedback on the approach shown there. It's unfortunately to lose the I/O methods but I can't think of a good solution immediately.

Reply:
Thanks for taking over, that's really appreciated. I do not have any better idea regarding to the error type than what's already implemented.